src/services/chain/messages/evm/transfer.ts

```diff
@@ -58,7 +58,61 @@ export class EvmTransferMessage implements ChainMessage<SendToolParams> {
     return InProgressTxDisplay;
   }
 
-  validate(params: SendToolParams, walletStore: WalletStore): boolean {
+  private async validateBalance(
+    params: {
+      denom: string;
+      amount: string;
+      chainName: string;
+    },
+    walletStore: WalletStore,
+  ): Promise<boolean> {
+    try {
+      const { denom, amount, chainName } = params;
+      const { erc20Contracts, nativeToken, rpcUrls, nativeTokenDecimals } =
+        chainRegistry[this.chainType][chainName];
+
+      const rpcProvider = new ethers.JsonRpcProvider(rpcUrls[0]);
+      const address = walletStore.getSnapshot().walletAddress;
+
+      if (denom.toUpperCase() === nativeToken) {
+        const rawBalance = await rpcProvider.getBalance(address);
+        const formattedBalance = ethers.formatUnits(
+          rawBalance,
+          nativeTokenDecimals,
+        );
+        return Number(formattedBalance) >= Number(amount);
+      }
+
+      const erc20Record = getERC20Record(denom, erc20Contracts);
+      if (!erc20Record) {
+        return false;
```

Comment:
a more descriptive error thrown here would be more useful, otherwise we would fall back to this generic error here https://github.com/Kava-Labs/oros/pull/201/files#diff-e49e9ee75a1f24a197cabb5cc3d100526d6f540e07558642fabc92d93adc88dcR126